lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -3008,19 +3077,21 @@ SPIRVValue *LLVMToSPIRVBase::transIntrinsicInst(IntrinsicInst *II,
       AnnotSubj = cast<CastInst>(AnnotSubj)->getOperand(0);
     }
 
-    IntelFPGADecorations Decorations;
-    if (BB->getModule()->isAllowedToUseExtension(
-            ExtensionID::SPV_INTEL_fpga_memory_attributes) ||
-        BB->getModule()->isAllowedToUseExtension(
-            ExtensionID::SPV_INTEL_fpga_memory_accesses))
-      // If it is allowed, let's try to parse annotation string to find
-      // IntelFPGA-specific decorations
-      // TODO: The check is not entirely correct, since it allows usage
-      // of extension A even if extension B is the only one enabled.
-      // With the existing stack, these are always enabled/disabled
-      // simultaneously, however we should find a way to separate the
-      // actions for each individual extension.
-      Decorations = tryParseIntelFPGAAnnotationString(AnnotationString);
+    std::string AnnotationString;
+    // Process main annotation with the attached optional integer parameters
+    if (GetElementPtrInst *GEP =
+            dyn_cast<GetElementPtrInst>(II->getArgOperand(1))) {
+      if (Constant *C = dyn_cast<Constant>(GEP->getOperand(0))) {
+        StringRef StrRef;
+        getConstantStringInfo(C, StrRef);
+        AnnotationString += StrRef.str();
+      }
+    }
+    if (BitCastInst *Cast = dyn_cast<BitCastInst>(II->getArgOperand(4)))
+      if (Constant *C = dyn_cast_or_null<Constant>(Cast->getOperand(0)))
+        processOptionalAnnotationInfo(C, AnnotationString);
+    AnnotationDecorations Decorations =
+        tryParseAnnotationString(BM, AnnotationString);
```

Comment:
Sure

Reply:
I probably left this code inlined because at some point I wanted to resolve:
```
Filename annotation is not being always and properly translated. To fix this additional
round of refactoring is required, yet again it will be done with the new extension implementation.
```
issue, and draft code was introducing some differences between these two blocks. But in the end I won't fix the issue in this PR and hence the code can and should be outlined in a different function.

Reply:
Applied. Yet I left
```
    AnnotationDecorations Decorations =
        tryParseAnnotationString(BM, AnnotationString);
```

outside of this function, since I'd expect these string to be unique for var, ptr and global annotations at some point.